kernel/aster-nix/src/fs/utils/page_cache.rs

```diff
@@ -140,6 +319,52 @@ impl PageCacheManager {
 
         Ok(())
     }
+
+    fn ondemand_readahead(&self, idx: usize) -> Result<Frame> {
+        let mut pages = self.pages.lock();
+        let mut ra_state = self.ra_state.lock();
+        let backend = self.backend();
+        // Check for the previous readahead.
+        ra_state.wait_for_prev_readahead(&mut pages, false)?;
+        // Three possible conditions on reaching here.
```

Comment:
There are three possible conditions that could be encountered upon reaching here.